src/components/CharacterAssignment.jsx

```diff
@@ -0,0 +1,229 @@
+        newColumns[assignedActor].lineCount += fetchedCharacters[key].lineCount
+      } else {
+        newColumns.unassignedCharacters.characterIds.push(fetchedCharacters[key].id)
+        newColumns.unassignedCharacters.lineCount += fetchedCharacters[key].lineCount
+      }
+    }
+    setColumns({...newColumns})
+  }
+
+  // Create and update the columns for each of the actors and an Unassigned Characters column
+  const createOrUpdateActorColumns = (actors) => {
+    // Script is already set, only add or remove actors
+    if (currentScript && actors.length > 1  ) {
+      const newColumns = {...columns}
+      const newOrder = [...columnOrder]
+      // Adding an actor, there are more actors than columns (plus the unassigned column)
+      if (actors.length + 1 > newOrder.length){
+        actors.forEach((actor) => {
+          // Adding a new actor
+          if (!newColumns[actor.id]){
+            newColumns[actor.id] = {
+              id: actor.id.toString(),
+              title: actor.name,
+              characterIds: [],
+              lineCount: 0
+            }
+          newOrder.push(actor.id)
+          }
+        })
+      }
+      // Removing an actor, there are more columns than actors
+      else {
+        for (let key in newColumns){
+          // The current column doesn't match any actor or the unassigned column
+          if (!actors.some((actor) => actor.id == key) && key != 'unassignedCharacters'){
+          // Reassign characterIds to the unassigned column
+            const reassignedIds = newColumns[key].characterIds;
+            const reassignedCounts = newColumns[key].lineCount;
+            newColumns.unassignedCharacters.characterIds = [...newColumns.unassignedCharacters.characterIds, ...reassignedIds];
+            newColumns.unassignedCharacters.lineCount += reassignedCounts;
+            const index = newOrder.findIndex(id => id === key);
+            newOrder.splice(index, 1);
+          }
+        };
+      }
+      //Update the new column and order values after the actor changes
+      setColumns(newColumns)
+      setColumnOrder(newOrder)
+    }
+
+    // Initial setup of actor columns, no script is selected
+    else {
+    const actorColumns = {
+      'unassignedCharacters': {
+        id:'unassignedCharacters',
+        title: 'Unassigned Characters',
+        characterIds: [],
+        lineCount: 0
+      }
+    };
+    const actorIds = ['unassignedCharacters']
+    actors.forEach((actor) => {
+      actorColumns[actor.id] = {
+        id: actor.id.toString(),
+        title: actor.name,
+        characterIds: [],
+        lineCount: 0
+      }
+      actorIds.push(actor.id.toString())
+    })
+      setColumns(actorColumns)
+      setColumnOrder(actorIds)
+    }
+  }
+
+
+  // Gets called after a character is dropped on a new actor or unassigned. Updates the value in the database.
+  const addCharacter = async (characterId, actorId, currentScript) => {
```

Comment:
Like the other comment on the repo file, maybe rename this to something more precise such as "assignCharacter"?  
Also, should `currentScript` be called `scriptId` or `currentScriptId` to be consistent with the other parameters?